learn/guides/calling-java-code-from-ballerina.md

```diff
@@ -0,0 +1,290 @@
+intro: Ballerina offers a straightforward way to call the existing Java code from Ballerina. Although Ballerina is not designed to be a JVM language, the current implementation, which targets the JVM, aka jBallerina, provides Java interoperability by adhering to the Ballerina language semantics.
+redirect_from:
+  - /learn/how-to-use-java-interoperability
+  - /learn/how-to-use-java-interoperability/
+  - /learn/how-to-call-java-code-from-ballerina
+  - /learn/how-to-call-java-code-from-ballerina/
+  - /learn/calling-java-code-from-ballerina
+  - /swan-lake/learn/calling-java-code-from-ballerina
+  - /swan-lake/learn/calling-java-code-from-ballerina/
+  - /learn/user-guide/calling-java-code-from-ballerina
+  - /learn/user-guide/calling-java-code-from-ballerina/
+  - /learn/user-guide/interoperability/calling-java-code-from-ballerina
+  - /learn/user-guide/interoperability/calling-java-code-from-ballerina/
+  - /learn/user-guide/interoperability/
+  - /learn/user-guide/interoperability
+  - /learn/calling-java-code-from-ballerina-and-vice-versa
+  - /learn/calling-java-code-from-ballerina-and-vice-versa/
+---
+
+## Writing Ballerina Bindings
+We have introduced a CLI tool called `bindgen` to ease the process of generating Ballerina bindings for given Java APIs. This guide shows you how to use it, if you need more details on how the bindgen tool works, you can refer its reference guide [here](/learn/java-interoperability/the-bindgen-tool/).
+
+The [Package Layout Guide](/learn/organizing-ballerina-code/package-layout/) explains how to package Java libraries (JAR files) with Ballerina programs. When you generate bindings for a Java library using the bindgen tool, this part is already handled. However, you could gain an in-depth understanding of how Java libraries and their transitive dependencies are packaged with Ballerina programs to produce self-contained executable programs.
+
+The [Ballerina FFI Guide](/learn/java-interoperability/ballerina-ffi/) explains how to write these bindings manually. It is also a useful section for those who want to understand the inner workings of calling Java from Ballerina and for those who want to customize the bindings generated by the bindgen tool.
+
+## Using the SnakeYAML Java Library in Ballerina
+SnakeYAML is a YAML parser for Java. In this section, we'll learn how to use this library to parse a YAML document using Ballerina. 
+
+We'll develop a Ballerina program that parses the given YAML file and writes the content to the standard out.
+
+Let's get started.
+
+### Step 1 - Writing the Java Code
+It is recommended to always start by writing the Java code. It gives you an idea of the set of Java classes required to implement your logic. Then, we can use the `bindgen` tool to generate Ballerina bindings for those classes. 
+
+The Java code below uses the SnakeYAML API to parse the given YAML file. Note that this is not the most idiomatic way of writing the Java code for this scenario. 
```

Comment:
```suggestion
The Java code below uses the SnakeYAML API to parse the given YAML file. 

>**Note:** This is not the most idiomatic way of writing the Java code for this scenario. 
```

Reply:
Should we explain the most idiomatic way also if this is not it?

Reply:
No, it is a Java specific thing that we don't need to get into details.